src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue

```diff
@@ -270,17 +270,16 @@ export default {
 		}"
 		:style="cssVars"
 		class="checkbox-radio-switch">
-		<input :id="id"
-			:checked="isChecked"
+		<input :checked="isChecked"
 			:disabled="disabled"
 			:indeterminate="indeterminate"
 			:name="name"
 			:type="inputType"
 			:value="value"
 			class="checkbox-radio-switch__input"
 			@change="onToggle">
-		<label :for="id" class="checkbox-radio-switch__label">
-			<div class="checkbox-radio-switch__icon">
+		<button class="checkbox-radio-switch__button" @click="onToggle">
```

Comment:
`<label>` cannot receive any of the events in the class of `focus*` hence the only way to make this focus-able it use an element that can give us that and here a button seems appropriate.

Buttons, get some focus already however unpredictable but that would be addressed in another issue here : https://github.com/nextcloud/nextcloud-vue/issues/2824

Reply:
Sorry, I understand the reasoning, but label+input semantics are much more important than a focus-visible :)

Using keyboard, the focus-visible will be on th einput.
Which allows you to style the label by using a css direct sibling selector: `input:focus-visible + label`